addons/mail/static/src/models/channel.js

```diff
@@ -0,0 +1,95 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one, many } from '@mail/model/model_field';
+import { insertAndReplace } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'Channel',
+    identifyingFields: ['id'],
+    recordMethods: {
+        async fetchChannelMembers() {
+            const channelData = await this.messaging.rpc({
+                model: 'mail.channel',
+                method: 'load_more_members',
+                args: [[this.id]],
+                kwargs: {
+                    known_member_ids: this.channelMembers.map(channelMember => channelMember.id),
+                },
+            });
+            if (!this.exists()) {
+                return;
+            }
+            this.update(channelData);
+        },
+        /**
+         * @private
+         * @returns {Boolean|FieldCommand}
+         */
+        _computeAreAllMembersLoaded() {
+            return this.memberCount === this.channelMembers.length;
+        },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeThread() {
+            return insertAndReplace({
+                id: this.id,
+                model: 'mail.channel',
+            });
+        },
+        _sortMembers() {
+            return [
+                ['defined-first', 'name'],
+                ['case-insensitive-asc', 'name'],
+            ];
+        },
+        /**
+         * @private
+         * @returns {integer}
+         */
+        _computeUnknownMemberCount() {
+            return this.memberCount - this.channelMembers.length;
+        },
+    },
+    fields: {
+        areAllMembersLoaded: attr({
+            compute: '_computeAreAllMembersLoaded',
+        }),
+        channelMembers: many('ChannelMember', {
+            inverse: 'channel',
+            isCausal: true,
+        }),
+        id: attr({
+            readonly: true,
+            required: true,
+        }),
+        memberCount: attr({
+            related: 'thread.memberCount',
```

Comment:
could be moved here, but I guess ok to do it in a next step too

Reply:
It's not that easy to move since it could be filled by the `thread.convertData`. This is why I used a related instead of moving the field.

Reply:
Ok then it's a problem for another time.